source/application/SXRMB/SXRMBAppController.cpp

```diff
@@ -76,13 +77,10 @@ bool SXRMBAppController::startup()
 	// Start up the main program.
 	if(AMAppController::startup()) {
 
-
 		// Initialize central beamline object
 		SXRMBBeamline::sxrmb();
 		// Initialize the periodic table object.
 		AMPeriodicTable::table();
-		// Initialize the storage ring
-		CLSStorageRing::sr1();
```

Comment:
@XuetaoLiu-CLS 
CLSStorageRing::sr1() is just a singleton class ... so it can be called from anywhere. It will just get created the first time it's called, so it doesn't really matter if that's in the appController or the beamline class. I might suggest for the sake of consistency that you bring a call back into the appController here and just put it before the SXRMBBeamline::sxrmb() call. But if you want to redefine what you're considering beamAvailable(), then you've done the right thing by leaving it to the beamline class.